main_app/views.py

```diff
@@ -1,13 +1,13 @@
-from django.shortcuts import render, redirect
+from django.shortcuts import render, redirect, get_object_or_404
 from django.views.generic.edit import CreateView, UpdateView, DeleteView
-from .models import Item, Category
+from .models import Item, Category, HistoryLog
 from django.contrib.auth import authenticate, login, logout
 from django.contrib import messages
 from django.contrib.auth.decorators import login_required
 from .forms import SignUpForm
+from django.utils.decorators import method_decorator
```

Comment:
Just in case you guys want to safeguard these functions using @login_required or something